edgexfoundry/app-record-replay
# feat: Implement app layer Record Status

closes #28







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/app-record-replay/blob/main/.github/CONTRIBUTING.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have opened a PR for the related docs change (if not, why?) **TBD**
  <link to docs PR>

## Testing Instructions
Run non-secure EdgeX stack
Build app service
run `./app-record-replay -cp -d | grep -e "ARR "
Send **GET** to `localhost:59712/api/v3/record`
Verify results are:
```
{
    "inProgress": false,
    "eventCount": 0,
    "duration": 0
}
```
Send **POST** to `localhost:59712/api/v3/record` with
```
{
    "duration" : 60000000000,
    "eventLimit" : 10
}
```
After 10 seconds Send **GET** to `localhost:59712/api/v3/record`
Verify results are:
```
{
    "inProgress": true,
    "eventCount": {non-zero value},
    "duration": {non-zero value)
}
```
After in a minute or so Send **GET** to `localhost:59712/api/v3/record`
Verify results are:
```
{
    "inProgress": false,
    "eventCount": 10,
    "duration": {non-zero value less than 60000000000)
}
```
Send **POST** to `localhost:59712/api/v3/record` with
```
{
    "duration" : 6000000000,
    "eventLimit" : 100
}
```
After in a 25 seconds, send **GET** to `localhost:59712/api/v3/record`
Verify results are:
```
{
    "inProgress": false,
    "eventCount": {value less than 100},
    "duration": {value greater than 6000000000}
}
```


## New Dependency Instructions (If applicable)
